diff --git a/spec/temporal_expr_spec.rb b/spec/temporal_expr_spec.rb
index 7c54fb7..13286d9 100644
--- a/spec/temporal_expr_spec.rb
+++ b/spec/temporal_expr_spec.rb
@@ -82,5 +82,35 @@ describe "temporal expression" do
 
   context "range every year temporal expr을 정의할 수 있다" do
     # 시작과 끝에 해당하는 day, month를 인자로 받는다
+
+    class RangeEachYear < TemporalExpr
+      def initialize stt_mon, end_mon, stt_day=0, end_day=0
+        @stt_mon, @end_mon = stt_mon, end_mon
+        @stt_day, @end_day = stt_day, end_day
+      end
+
+      def includes date
+        months_include date or stt_month_includes date or end_month_includes date
+      end
+
+      def months_include date
+        mon = date.mon
+        mon > @stt_mon and mon < @end_mon
+      end
+
+      def stt_month_includes date
+        false if date.mon != @stt_mon
+        true if @stt_day == 0
+        date.day >= @stt_day
+      end
+
+      def end_month_includes date
+        false if date.mon != @end_mon
+        true if @end_day == 0
+        date.day <= @end_day
+      end
+    end
+
+    range_each_year = RangeEachYear.new(3, 9, 14, 12)
   end
 end
\ No newline at end of file
